### DISCUSSION TITLE
Using charm on subjects with defacing

### DISCUSSION THREAD

**Discussion by michefabb** [simnibs/simnibs#428] [7.Creating head models] [answered]
Dear SimNIBS experts,

I am trying to reconstruct the head model using charm with T1w and T2w of defaced subjects from HBN database. At first attempt it is not optimal, as the frontal part of brain is cut and (even a dummy) skull is not built, see below.
![image](https://github.com/user-attachments/assets/517e97c4-ff25-4903-b00d-6b3c18c1c578)

I am using this together with dwi2cond to calculate the Leadfield Matrix for EEG forward calculation, and this would be problematic.

Would you kindly suggest any possible strategy to improve?

I was also noting that even if the MRI is defaced as below, charm partially reconstructs the defaced part (maybe from its templates?).
![image](https://github.com/user-attachments/assets/89ef3163-1c91-447f-ba40-9d6564ad5627)

Thanks, appreciated,

Mic

**Comment by bjornsig** [comment]
Hi Mic,

It could be that the initial registration is failing. Usually charm puts some default face on defaced images. You can check the charm html report to see the initial registration. It is located in the m2m folder, charm_report.html.

You can try to run charm with the --no_neck flag.

**Reply by michefabb** [reply]
Thanks @bjornsig ,

you can see here the pic of the report with default settings:
![Screenshot 2024-09-27 alle 14 25 48](https://github.com/user-attachments/assets/65fba5f0-6ef5-4b8c-99df-2ea0dab752d8)
![Screenshot 2024-09-26 alle 16 48 17](https://github.com/user-attachments/assets/33d7a9eb-9261-4aca-864c-b06f8d4b309c)

I tried changing the rotation in the settings.ini file to `affine_rotations = [0]` but the result is not good (see below):
![Screenshot 2024-09-27 alle 14 26 44](https://github.com/user-attachments/assets/9490e7b8-7c94-4d78-8221-209495f848eb)
![Screenshot 2024-09-26 alle 16 50 24](https://github.com/user-attachments/assets/e27df26f-0163-4ff5-a526-ea89473dac74)
So I tried using `--noneck` flag with default settings and it improved a bit (see below) but did not fix completely:
![Screenshot 2024-09-27 alle 14 27 18](https://github.com/user-attachments/assets/2d6a5c43-a2fe-4e32-87a6-815def95a627)
![Screenshot 2024-09-26 alle 16 52 53](https://github.com/user-attachments/assets/749a3468-f0c2-4e5f-a1c7-0adabdb6c002)

Any suggestion about how I could improve? Maybe --noneck and also changing the settings.ini for affine registration? Would you kindly suggest how?

Thanks, appreciated your support!

Mic

**Reply by oulap** [reply]
Hi Mic,

Can you play around with the affine_rotations a bit more? I.e., set it to affine_rotations = [-15] for one run and affine_rotations = [15] for another, and then re-check the template.

If that doesn't help, you can also create the affine registration using and external tool and pass the transformation to charm.

**Reply by grodgers1** [reply, marked as answer]
Hi, just to add in from my experience (though I haven't used de-faced images): it can really help to manually initialize the affine registration, see comment here about how to do this practically: https://github.com/simnibs/simnibs/discussions/172.

**Reply by michefabb** [reply]
Thanks @oulap @grodgers1 ,

@grodgers1 thanks for your post, I am struggling to find where is the SimNIBS template... is it `/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/segmentation/atlases/charm_atlas_mni/template.nii` ?

**Reply by grodgers1** [reply]
Yes, that one, also found [here](https://github.com/simnibs/simnibs/blob/master/simnibs/segmentation/atlases/charm_atlas_mni/template.nii)

**Reply by michefabb** [reply]
Thanks @grodgers1 , using your post and the --noneck flag as suggested by @bjornsig worked! Thanks all for supporting!

Mic
